plugins/network-elements/ovs/src/com/cloud/network/element/OvsElement.java

```diff
@@ -488,50 +494,54 @@ public boolean applyPFRules(final Network network, final List<PortForwardingRule
             return true;
         }
 
-        DataCenterVO dcVO = _dcDao.findById(network.getDataCenterId());
-        NetworkTopology networkTopology = _networkTopologyContext.retrieveNetworkTopology(dcVO);
-
-        return networkTopology.applyFirewallRules(network, rules, routers);
+        final DataCenterVO dcVO = _dcDao.findById(network.getDataCenterId());
+        final NetworkTopology networkTopology = _networkTopologyContext.retrieveNetworkTopology(dcVO);
+        for (final DomainRouterVO domainRouterVO : routers) {
+            result = networkTopology.applyFirewallRules(network, rules, domainRouterVO);
```

Comment:
Sorry, i did not notice. Where did you say that?
I saw some other places where you bail on the first fail. I don't like that solution:
If the first fails nothing is applied.
If the second fails the first router has the rules applied.
with &= it is symmetrical.